src/operator/debounce.ts

```diff
@@ -8,6 +8,13 @@ import {tryCatch} from '../util/tryCatch';
 import {isPromise} from '../util/isPromise';
 import {errorObject} from '../util/errorObject';
 
+/**
+ * Returns an Observable the same as source Observable, but drops items
+ * emitted by the source followed by additional items before the computed
+ * timeout based on each item in the source.
```

Comment:
For debounce(Time), to keep it as straight-to-the-point as possible, I'd describe it as
"*Returns the source Observable delayed by the computed debounce duration, except it extends the duration if a new source item arrives before the previous source item was emitted on the result Observable. In practice, for each item emitted on the source, this operator holds the latest item, waits for 'silence' as long as the `durationSelector` specifies, and only then emits the latest source item on the result Observable.*"